connections/evm/connection.go

```diff
@@ -22,76 +22,86 @@ import (
 	"github.com/ethereum/go-ethereum/rpc"
 )
 
+type connection interface { //nolint:deadcode,unused
+	Connect() error
+	Close()
+	NetworkId() (*big.Int, error)
+	LatestBlock() (*big.Int, error)
+	EnsureHasBytecode(addr ethcommon.Address) error
+	NewTransactOpts(value, gasLimit, gasPrice *big.Int) (*bind.TransactOpts, uint64, error)
+	WaitForBlock(block *big.Int) error
+}
+
 type Connection struct {
-	cfg       Config
-	ctx       context.Context
-	conn      *ethclient.Client
-	signer    ethtypes.Signer
-	kp        *secp256k1.Keypair
-	nonceLock sync.Mutex
+	Cfg       Config
+	Ctx       context.Context
+	Conn      *ethclient.Client
+	Signer    ethtypes.Signer
+	Kp        *secp256k1.Keypair
+	NonceLock sync.Mutex
 	log       log15.Logger
-	stop      <-chan int // All routines should exit when this channel is closed
+	Stop      <-chan int // All routines should exit when this channel is closed
```

Comment:
If other components depend on these they need to have getter methods in the interface, they shouldn't be exported here.

Pretty certain not all of these needed elsewhere. I think just `Kp`, `NonceLock` and maybe some stuff in the config are required.